File: app/src/main/java/com/shopcart/Activities/MainActivity/Fragments/MenuFragment.java
package com.shopcart.Activities.MainActivity.Fragments;


import android.app.Activity;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.navigation.NavController;
import androidx.navigation.NavDestination;
import androidx.navigation.NavDirections;
import androidx.navigation.fragment.NavHostFragment;

import com.google.firebase.auth.FirebaseAuth;
import com.shopcart.R;
import com.shopcart.Utilities.Tasks;
import com.shopcart.databinding.FragmentMenuBinding;

/**
 * A simple {@link Fragment} subclass.
 */
public class MenuFragment extends Fragment implements View.OnClickListener {
    private FragmentMenuBinding binding;
    private FragmentManager fragmentManager;
    private Activity activity;

    FirebaseAuth firebaseAuth;

    public MenuFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_menu, container, false);
        if (isAdded()) activity = getActivity();

        fragmentManager = getFragmentManager();

        firebaseAuth = FirebaseAuth.getInstance();

        binding.menuBtnHome.setOnClickListener(this);
        binding.menuBtnProfile.setOnClickListener(this);
        binding.menuBtnCart.setOnClickListener(this);
        binding.menuBtnFavourite.setOnClickListener(this);
        binding.menuBtnOrders.setOnClickListener(this);
        binding.menuBtnLanguage.setOnClickListener(this);
        binding.menuBtnSettings.setOnClickListener(this);
        binding.menuImgClose.setOnClickListener(this);
        binding.menuBtnSignOut.setOnClickListener(this);

        return binding.getRoot();
    }

    @Override
    public void onClick(View v) {
        if (v.equals(binding.menuBtnHome)) {
            activity.onBackPressed();

        } else if (v.equals(binding.menuBtnProfile)) {
            NavDirections action = MenuFragmentDirections.actionMenuFragmentToProfileFragment();
            NavController navController = NavHostFragment.findNavController(this);
            NavDestination navDestination = navController.getCurrentDestination();
            if (navDestination != null && navDestination.getId() == R.id.menuFragment)
                navController.navigate(action);

        } else if (v.equals(binding.menuBtnCart)) {
            NavDirections action = MenuFragmentDirections.actionMenuFragmentToCartFragment();
            NavController navController = NavHostFragment.findNavController(this);
            NavDestination navDestination = navController.getCurrentDestination();
            if (navDestination != null && navDestination.getId() == R.id.menuFragment)
                navController.navigate(action);


        } else if (v.equals(binding.menuBtnFavourite)) {
            NavDirections action = MenuFragmentDirections.actionMenuFragmentToFavouriteFragment();
            NavController navController = NavHostFragment.findNavController(this);
            NavDestination navDestination = navController.getCurrentDestination();
            if (navDestination != null && navDestination.getId() == R.id.menuFragment)
                navController.navigate(action);


        } else if (v.equals(binding.menuBtnOrders)) {
            NavDirections action = MenuFragmentDirections.actionMenuFragmentToMyOrdersFragment();
            NavController navController = NavHostFragment.findNavController(this);
            NavDestination navDestination = navController.getCurrentDestination();
            if (navDestination != null && navDestination.getId() == R.id.menuFragment)
                navController.navigate(action);


        } else if (v.equals(binding.menuBtnLanguage)) {
            NavDirections action = MenuFragmentDirections.actionMenuFragmentToLanguageFragment();
            NavController navController = NavHostFragment.findNavController(this);
            NavDestination navDestination = navController.getCurrentDestination();
            if (navDestination != null && navDestination.getId() == R.id.menuFragment)
                navController.navigate(action);

        } else if (v.equals(binding.menuBtnSettings)) {
            NavDirections action = MenuFragmentDirections.actionMenuFragmentToSettingsFragment();
            NavController navController = NavHostFragment.findNavController(this);
            NavDestination navDestination = navController.getCurrentDestination();
            if (navDestination != null && navDestination.getId() == R.id.menuFragment)
                navController.navigate(action);


        }else if (v.equals(binding.menuBtnSignOut)) {
            Tasks.signOut(activity, firebaseAuth);
            activity.onBackPressed();

        } else if (v.equals(binding.menuImgClose)) {
            activity.onBackPressed();
        }
    }


}

File: app/src/main/java/com/shopcart/ui/fragments/MenuFragment.kt
package com.shopcart.ui.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.NavHostFragment
import com.shopcart.R
import com.shopcart.databinding.FragmentMenuBinding
import com.shopcart.ui.viewModels.MainViewModel
import dagger.hilt.android.AndroidEntryPoint
import es.dmoral.toasty.Toasty

private const val TAG = "Menu Fragment"

@AndroidEntryPoint
class MenuFragment : Fragment(), View.OnClickListener {
    private lateinit var binding: FragmentMenuBinding
    private val viewModel: MainViewModel by viewModels()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_menu, container, false)

        binding.apply {
            menuBtnHome.setOnClickListener(this@MenuFragment)
            menuBtnProfile.setOnClickListener(this@MenuFragment)
            menuBtnCart.setOnClickListener(this@MenuFragment)
            menuBtnFavourite.setOnClickListener(this@MenuFragment)
            menuBtnOrders.setOnClickListener(this@MenuFragment)
            menuBtnLanguage.setOnClickListener(this@MenuFragment)
            menuBtnSettings.setOnClickListener(this@MenuFragment)
            menuImgClose.setOnClickListener(this@MenuFragment)
            menuBtnSignOut.setOnClickListener(this@MenuFragment)
        }

        return binding.root
    }

    override fun onClick(v: View) {
        when (v) {
            binding.menuBtnHome -> requireActivity().onBackPressed()

            binding.menuBtnProfile -> {
                openProfileFragment()
            }

            binding.menuBtnCart -> {
                openCartFragment()
            }

            binding.menuBtnFavourite -> {
                openFavouriteFragment()
            }

            binding.menuBtnOrders -> {
                openMyOrdersFragment()
            }

            binding.menuBtnLanguage -> {
                openLanguageFragment()
            }

            binding.menuBtnSettings -> {
                openSettingsFragment()
            }

            binding.menuBtnSignOut -> {
                viewModel.signOut()
                //Show Toast
                Toasty.success(activity!!, "signed out", Toast.LENGTH_SHORT, true).show()
                openWelcomeFragment()

            }
            binding.menuImgClose ->
                requireActivity().onBackPressed()
        }
    }


    private fun openProfileFragment() {
        val action = MenuFragmentDirections.actionMenuFragmentToProfileFragment()
        val navController = NavHostFragment.findNavController(this)
        val navDestination = navController.currentDestination
        if (navDestination != null && navDestination.id == R.id.menuFragment) navController.navigate(
            action
        )
    }

    private fun openCartFragment() {
        val action = MenuFragmentDirections.actionMenuFragmentToCartFragment()
        val navController = NavHostFragment.findNavController(this)
        val navDestination = navController.currentDestination
        if (navDestination != null && navDestination.id == R.id.menuFragment) navController.navigate(
            action
        )
    }

    private fun openFavouriteFragment() {
        val action = MenuFragmentDirections.actionMenuFragmentToFavouriteFragment()
        val navController = NavHostFragment.findNavController(this)
        val navDestination = navController.currentDestination
        if (navDestination != null && navDestination.id == R.id.menuFragment) navController.navigate(
            action
        )
    }

    private fun openMyOrdersFragment() {
        val action = MenuFragmentDirections.actionMenuFragmentToMyOrdersFragment()
        val navController = NavHostFragment.findNavController(this)
        val navDestination = navController.currentDestination
        if (navDestination != null && navDestination.id == R.id.menuFragment) navController.navigate(
            action
        )
    }

    private fun openLanguageFragment() {
        val action = MenuFragmentDirections.actionMenuFragmentToLanguageFragment()
        val navController = NavHostFragment.findNavController(this)
        val navDestination = navController.currentDestination
        if (navDestination != null && navDestination.id == R.id.menuFragment) navController.navigate(
            action
        )
    }

    private fun openSettingsFragment() {
        val action = MenuFragmentDirections.actionMenuFragmentToSettingsFragment()
        val navController = NavHostFragment.findNavController(this)
        val navDestination = navController.currentDestination
        if (navDestination != null && navDestination.id == R.id.menuFragment) navController.navigate(
            action
        )
    }

    private fun openWelcomeFragment() {
        val action = MenuFragmentDirections.actionMenuFragmentToEntryFragments()
        val navController = NavHostFragment.findNavController(this)
        val navDestination = navController.currentDestination
        if (navDestination != null && navDestination.id == R.id.menuFragment) navController.navigate(
            action
        )
    }

}